local/blockchain.go

```diff
@@ -751,56 +751,27 @@ func (ln *localNetwork) createBlockchainConfigFiles(
 	log.Info(logging.Green.Wrap("creating config files for each custom chain"))
 	for i, chainSpec := range chainSpecs {
 		chainAlias := blockchainTxs[i].ID().String()
-
-		// create config, network upgrade and subnet config files
-		if chainSpec.ChainConfig != nil {
+		// update config info. set defaults and node specifics
+		if chainSpec.ChainConfig != nil || len(chainSpec.PerNodeChainConfig) != 0 {
 			created = true
+			ln.chainConfigFiles[chainAlias] = string(chainSpec.ChainConfig)
 			for nodeName := range ln.nodes {
-				nodeRootDir := getNodeDir(ln.rootDir, nodeName)
-				chainConfigDir := filepath.Join(nodeRootDir, chainConfigSubDir)
-				chainConfigPath := filepath.Join(chainConfigDir, chainAlias, configFileName)
-				if err := createFileAndWrite(chainConfigPath, chainSpec.ChainConfig); err != nil {
-					return false, fmt.Errorf("couldn't write chain config file at %q: %w", chainConfigPath, err)
+				if cfg, ok := chainSpec.PerNodeChainConfig[nodeName]; ok {
+					ln.nodes[nodeName].config.ChainConfigFiles[chainAlias] = string(cfg)
+				} else {
+					delete(ln.nodes[nodeName].config.ChainConfigFiles, chainAlias)
 				}
 			}
 		}
 		if chainSpec.NetworkUpgrade != nil {
 			created = true
-			for nodeName := range ln.nodes {
```

Comment:
why don't we need all of this deleted code anymore?

Reply:
That code was creating the files for chain config, network upgrade, subnet config for each node previous to network restart. Also was setting network/node config data structures so snapshot persistence works. But... I realized that network restart already takes care of creating such files, if the network/node config data structures are defined. So only keep that last part. Also avoiding duplicate code for creation of those files.
Also tested that all continues working, for all involved files.